src/ekf.cpp

```diff
@@ -173,9 +174,10 @@ void Ekf::correct(const Measurement & measurement)
 
   // Wrap angles in the innovation
   for (size_t i = 0; i < update_size; ++i) {
-    if (update_indices[i] == StateMemberRoll ||
+    if ((update_indices[i] == StateMemberRoll ||
       update_indices[i] == StateMemberPitch ||
-      update_indices[i] == StateMemberYaw)
+      update_indices[i] == StateMemberYaw) &&
+      std::isfinite(innovation_subset(i)))
```

Comment:
So if the innovation has values that are not finite, we should look into why. Otherwise, this feels like we're trying to just cover up a bigger issue.

Also, oof. I need to update this code to use [this](https://github.com/ros/angles/blob/2392b18b4101cf2e81b77e89e3535959ad03d1ca/angles/include/angles/angles.h#L83).

Reply:
Probably is not your fault - I modified little a bit this code and BANG I got infinity loop. I think, this function should be protected against wrong values.

I will try to add "angles" library